index.js

```diff
@@ -115,3 +64,85 @@ app.on('activate', () => {
         createWindow();
     }
 });
+
+function getArgs() {
+    // Get the parameters
+    let failed = false;
+    var argv = yargs(hideBin(process.argv))
+        .help("h").alias("h", "help")
+        .usage("Usage: $0 [<options>]")
+        .example("$0 -b zepto.bin", "Start the emulator and execute zepto.bin in the emulator")
+        .option('rom', {
+            type: 'string',
+            alias: 'r',
+            description: 'Select a prebuild romdisk or a local image',
+            nargs: 1,
+        })
+        .option('breakpoint', {
+            type: 'string',
+            alias: 'B',
+            description: 'Set a breakpoint',
+            array: true,
+            nargs: 1,
+        })
+        .array("breakpoint")
+        .option('map', {
+            type: 'string',
+            alias: 'm',
+            description: 'Load a map file',
+            nargs: 1,
+        })
+        .option('eeprom', {
+            type: 'string',
+            alias: 'e',
+            description: 'Load a file into EEPROM',
+            nargs: 1,
+        })
+        .option('cf', {
+            type: 'string',
+            alias: 'c',
+            description: 'Load a file into CompactFlash',
+            nargs: 1,
+        })
+        .strictOptions()
+        .exitProcess(false)
+        .fail((msg, err, yargs) => {
+            console.log(msg);
+            // if(error) throw err;
+            console.log(yargs.help());
+            failed = true;
+        })
+        .parse();
+    if(failed) return false;
+    return argv;
+}
+
+function parseArgs(argv) {
+    if (argv.rom) {
+        let img_regex = /^v[0-9]*\.[0-9]*\.[0-9]*/;
```

Comment:
why do images need to meet this regex filename requirement? Can I not pass a custom image named `whatever-i-like.ext`?

If the file exists, we should be able to load it.

Reply:
Even if `argv.rom` doesn't meet the regex, it will still be read. If you have a image like `v0.4.0-9-ge68eb04.img` in current directory, use `-r ./v0.4.0-9-ge68eb04.img`.

Reply:
not sure i understand how the regex is useful ... as it appears to only partially match a version number ... so with `v0.4.0-9-ge68eb04.img` it only matches the `v0.4.0` portion ... if this matches, it prepends `roms/` ... this seems like "magic" and may create confusion.

if the user is passing CLI args, and the image is in the roms folder ... can they not just use auto-completion on the shell and type `-r ./roms/v0.4.0[TAB]` ... as this regex requires the user to know the full filename in the roms folder anyhow, since it still uses argv.rom ... so if I type `-r v0.4.0` it will fail, since the roms have the change and SHA included in their filenames?

If they type `-r v0.4.0` I would expect it to look for a file named `v0.4.0` in the current directory 🤷

Reply:
From what I understand, the assumption made here is that the ROM that will be loaded were downloaded from the remote pre-built images, for example, providing `v0.4.0` would load the previously-downloaded v0.4.0 release.

However, as @zoul0813 said, we want to be able to load any rom file, no matter what its name is. For example, the default file name generated by Zeal 8-bit OS is `os_with_romdisk.img`, it doesn't contain any `vX.Y.Z` string.
If the user wants to load a pre-downloaded ROM, I prefer that he specifies the relative path, including the leading `roms/`

Reply:
@zoul0813 No, `.test()` method returns a boolean. If **any part** of `argv.rom` meets this regex, the program will think that it is a remote image. What's more, if there's anything before the version number, it will return false.

Reply:
Removed👌